crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1142,11 +1163,6 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                     .iter()
                     .flat_map(|requirement| {
                         PubGrubDependency::from_requirement(requirement, None, fork_locals)
-                            // Keep track of local versions to propagate to transitive dependencies.
-                            .map_ok(|dependency| PubGrubDependency {
-                                local: locals::from_source(&requirement.source),
-                                ..dependency
-                            })
```

Comment:
Looks like there was a bug in https://github.com/astral-sh/uv/pull/5104 because we only looked for locals on the root package, which ignored editables and path dependencies. We have to set these fields in `PubGrubDependency::from_requirement` directly.

Reply:
Hmm this might be incorrect, but not exactly sure why.